numba/tests/test_config.py

```diff
@@ -137,6 +137,34 @@ def test_default_error_style_handling(self):
         ex_expected = "----> new_style"
         self.assertIn(ex_expected, out_msg, msg=err_msg)
 
+    @unittest.skipUnless(config.ENABLE_AVX,
+                         "test expects NUMBA_ENABLE_AVX==True")
+    def test_nocona_disables_avx(self):
+        # test with nocona
+        new_env = os.environ.copy()
+        new_env.pop('NUMBA_ENABLE_AVX', None)  # clear NUMBA_ENABLE_AVX
+
+        new_env['NUMBA_CPU_NAME'] = 'nocona'
+        code = ("from numba.core import config\n"
+                "print('---->', bool(config.ENABLE_AVX))\n"
+                "assert not config.ENABLE_AVX")
+        out, err = run_in_subprocess(dedent(code), env=new_env)
```

Comment:
Not suggesting changing anything now, especially as this is already RTM, but I think `@TestCase.run_in_subprocess` is a simpler way of doing this - it allows env vars for the subprocess to be set in the decorator args.